src/lein_monolith/plugin.clj

```diff
@@ -42,6 +43,51 @@
     subprojects))
 
 
+(defn- with-meta*
+  "Returns an object of the same type and value as obj, with map m as its
+  metadata if the object can hold metadata."
+  [obj m]
+  (if (instance? clojure.lang.IObj obj)
+    (with-meta obj m)
+    obj))
+
+
+(defn- string->path
+  [s]
+  (-> s io/file (.toPath)))
+
+
+(defn- relativize
+  [root path]
+  (let [root-path (string->path root)
+        path-path (string->path path)]
+    (str (if (.startsWith path-path root-path)
```

Comment:
This *could* cause problems if for some reason the monolith has paths that refer to directories that are nested inside other monolith projects, but I can't see a reason one might do that outside of erroneous usage. Perhaps this should also check the root-path + project directories to make sure that the path doesn't start with those more specific paths, and then opt out of relativizing in that case.